src/commands/deploy/deploy.ts

```diff
@@ -106,7 +107,14 @@ export async function deploy(actionContext: IActionContext, arg1?: vscode.Uri |
     }
 
     await node.runWithTemporaryDescription("Deploying...", async () => {
-        await appservice.deploy(nonNullValue(node).root.client, <string>deployPath, context);
+        try {
+            await appservice.deploy(nonNullValue(node).root.client, <string>deployPath, context);
+        } catch (error) {
+            if (failureMoreInfoSurvey(parseError(error), nonNullValue(siteConfig))) {
+                actionContext.errorHandling.suppressDisplay = true;
```

Comment:
This doesn't seem to work, but I can't tell if I've just got mismatched packages. I can step through one of the `callWithTelemetryAndErrorHandling` calls that deployment is wrapped in, but apparently there's another one that ignores the context and displays the popup anyway?

FWIW, my deployment is broken because I have `azure-mgmt` in my requirements.txt (which is not installable), in case someone wants to attempt a repro.

Reply:
The other popup is coming from here: https://github.com/microsoft/vscode-azuretools/blob/main/appservice/src/deploy/waitForDeploymentToComplete.ts#L82

The problem is that it's _also_ doing custom error handling (So that we can add a "View Output" button), meaning it actually pops up the message before you even catch the error here. My gut instinct is that `shouldPromptForSurvey` should return false if `errorHandling.suppressDisplay` is already set to `true`. That specific error case usually has helpful logs in the output, i.e. a pip failure related to `azure-mgmt`, so I think directing them to the output is more important than prompting for the survey

Reply:
Yes, you're right. That message is always going to win, so that's where this whole change should have been made.

I'll post on the thread.